DatadogTrace/Tests/OpenTelemetry/OTelSpanTests.swift

```diff
@@ -0,0 +1,220 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+import DatadogInternal
+
+@testable import DatadogTrace
+
+final class OTelSpanTests: XCTestCase {
+    func testSpanResourceNameDefault() {
+        let writeSpanExpectation = expectation(description: "write span event")
+        let core = PassthroughCoreMock(expectation: writeSpanExpectation)
+
+        // Given
+        let tracer: DatadogTracer = .mockWith(core: core)
+        let span = tracer.spanBuilder(spanName: "OperationName").startSpan()
+
+        // When
+        span.end()
+
+        // Then
+        waitForExpectations(timeout: 0.5, handler: nil)
+        let events: [SpanEventsEnvelope] = core.events()
+        XCTAssertEqual(events.count, 1)
+        let recordedSpan = events.first!.spans.first!
```

Comment:
suggestion/ For convenience we could declare `core.spans()` extension on `PassthroughCoreMock` and avoid repeating this dance in all tests.
```swift
let recordedSpan = try XCTUnwrap(core.spans().first)
```

Reply:
sure